Release bundler 2.2.0.rc.1 (#3608)
rubygems/rubygems: bundler/CHANGELOG.md

## 2.2.0 (May xx, 2020)

Highlights:

>   - Windows support. There's still gotchas and unimplemented features, but a Windows CI is now enforced.

Reviewer: ```suggestion
  - Windows support. There are still gotchas and unimplemented features, but a Windows CI is now enforced.
```

Improvements:

  - `bundle outdated` now prints output in columns for better readability [#4474](https://github.com/rubygems/bundler/pull/4474)
  - bundler's `release` rake task now prints a better message when not being logged in and trying to push a gem [#7513](https://github.com/rubygems/bundler/pull/7513)
  - `BUNDLE_APP_CONFIG` environment variable is now documented [#7563](https://github.com/rubygems/bundler/pull/7563)
  - Original exception is now reported when bundler fails to load OpenSSL [#7527](https://github.com/rubygems/bundler/pull/7527)
  - RVM specific instructions for recompiling ruby is no longer recommended when bundler fails to load OpenSSL [#7597](https://github.com/rubygems/bundler/pull/7597)
  - Improve resolver debugging out from resolver [#7589](https://github.com/rubygems/bundler/pull/7589) and [#7590](https://github.com/rubygems/bundler/pull/7590)
  - Clarify `bundle config --local` docs [#3408](https://github.com/rubygems/rubygems/pull/3408)

Bugfixes:

  - Fix `bundle pristine` removing gems with local overrides. Be conservative by printing a warning and skipping the removal [#7423](https://github.com/rubygems/bundler/pull/7423)

>   - Fix multiplaform resolution edge cases [#7522](https://github.com/rubygems/bundler/pull/7522), [#7554](https://github.com/rubygems/bundler/pull/7554) and [#7578](https://github.com/rubygems/bundler/pull/7578). TODO: [#7554](https://github.com/rubygems/bundler/pull/7554) WILL MOST LIKELY BE REVERTED BEFORE RELEASE, since it was fixed at the `compact_index` gem and thus correctly formatted data will be coming from the server

Reviewer: ```suggestion
  - Fix multiplatform resolution edge cases [#7522](https://github.com/rubygems/bundler/pull/7522), [#7554](https://github.com/rubygems/bundler/pull/7554) and [#7578](https://github.com/rubygems/bundler/pull/7578). TODO: [#7554](https://github.com/rubygems/bundler/pull/7554) WILL MOST LIKELY BE REVERTED BEFORE RELEASE, since it was fixed at the `compact_index` gem and thus correctly formatted data will be coming from the server
```

Typo.

Author: I just can't spell this properly :laughing:

  - Fix `gems.rb` files containing `:path` gems using relative paths not working when the app is packaged as a `jar` with `warbler` [#7614](https://github.com/rubygems/bundler/pull/7614). TODO: double check and report to jruby. The fix is a workaround for [underlying jruby issues](https://github.com/rubygems/bundler/issues/7598#issuecomment-578870789).
  - Fix config location edge case where if `BUNDLE_APP_CONFIG` is set to an absolute path like in official ruby docker images, and there's no Gemfile up in the directory hierarchy, bundler would end up using the default config location instead of the customized one [#7622](https://github.com/rubygems/bundler/pull/7622)
  - Fix error message about missing permissions recommending a deprecated command [#7633](https://github.com/rubygems/bundler/pull/7633)
  - Fix `init_gems_rb` setting being ignored by `bundle gem` [#7629](https://github.com/rubygems/bundler/pull/7629)

>   - Fix "unresolvable warning" being printed on `bundle install` of multipliplatform `gems.rb` files without lockfiles, multiplatform is now managed automatically [#7580](https://github.com/rubygems/bundler/pull/7580)

Reviewer: ```suggestion
  - Fix "unresolvable warning" being printed on `bundle install` of multiplatform `gems.rb` files without lockfiles, multiplatform is now managed automatically [#7580](https://github.com/rubygems/bundler/pull/7580)
```

Typo.

## 2.2.0 (May xx, 2020)

Highlights:

  - Windows support. There's still gotchas and unimplemented features, but a Windows CI is now enforced.

>   - Full multiplatform support. Bundler should now seamlessly handle multiplatform `gems.rb` files.

Reviewer 2: Should we be using `Gemfile` instead of `gems.rb`? Lots of people  I talk to about Bundler didn't know this was a thing, so I'm wondering if this may cause some confusion for users?

Author: This is just me trying to start raising awareness of `gems.rb` support by talking about it, I'll mention both :+1:.

Author: I changed it to mention both and also backfilled recent changes.
